staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -92,34 +93,55 @@ func validator(s *schema.Structural, isResourceRoot bool) *Validator {
 }
 
 // Validate validates all x-kubernetes-validations rules in Validator against obj and returns any errors.
-func (s *Validator) Validate(fldPath *field.Path, sts *schema.Structural, obj interface{}) field.ErrorList {
+func (s *Validator) Validate(fldPath *field.Path, sts *schema.Structural, obj interface{}, costBudget int64) (errs field.ErrorList, remainingBudget int64) {
+	remainingBudget = costBudget
 	if s == nil || obj == nil {
-		return nil
+		return nil, remainingBudget
 	}
 
-	errs := s.validateExpressions(fldPath, sts, obj)
+	errs, remainingBudget = s.validateExpressions(fldPath, sts, obj, remainingBudget)
+	if remainingBudget < 0 {
+		return errs, remainingBudget
```

Comment:
this is (potentially) short-circuiting checks below... how are we guaranteeing validation never reports success if we short-circuit because of runtime cost being exceeded? all the callers I see of this method outside this package are ignoring the remainingBudget return value, so they wouldn't catch a `[], -1` return

Reply:
I have updated to return whenever met cost exceeds error.